javascript/modules/client/src/client.ts

```diff
@@ -680,64 +708,81 @@ export class Client extends ClientCore implements IClient {
           // TODO update when new subgraph schema is deployed
           plugins: dao.packages.map(
             (
-              pkg: { pkg: { id: string; __typename: string } },
+              plugin: SubgraphPlugin,
             ): InstalledPluginListItem => {
               return {
-                instanceAddress: pkg.pkg.id,
-                id: pkg.pkg.__typename,
+                instanceAddress: plugin.pkg.id,
+                // TODO
+                // this should be the address of the parent plugin
+                // or the ens name, subgraph does not have this info
+                // at this moment so for now, the __typename
+                // will do the job to differentitate between
+                // plugins, however, this will eventually change
+                id: plugin.pkg.__typename + "plugin.dao.eth",
                 version: "0.0.1",
               };
             },
           ),
-        });
-      }
-      return daosDetails;
+        };
+      }));
+      // return daosDetails;
     } catch (err) {
-      throw new Error("Cannot fetch the dao details data from GraphQL");
+      throw new GraphQLError("dao");
     }
   }
 
   private async _getBalances(
-    daoAddress: string,
+    daoAddressorEns: string,
     _tokenAddresses: string[],
   ): Promise<AssetBalance[] | null> {
     // TODO check address
+    let address = daoAddressorEns;
+    if (!isAddress(address)) {
+      await this.web3.ensureOnline()
+      const provider = this.web3.getProvider();
+      if (!provider) {
+        throw new RequiredProviderError();
+      }
+      const resolvedAddress = await provider.resolveName(address);
+      if (!resolvedAddress) {
+        throw new InvalidAddressOrEnsError();
+      }
+      address = resolvedAddress;
+    }
     try {
       await this.graphql.ensureOnline();
       const client = this.graphql.getClient();
-      const { dao } = await client.request(QueryBalances, {
-        address: daoAddress,
+      const { balances } = await client.request(QueryBalances, {
+        address,
       });
-      if (!dao) {
-        return null;
-      }
-      if (dao.balances.length === 0) {
+      if (balances.length === 0) {
         return [];
       }
-      const balances: AssetBalance[] = [];
-      for (let i = 0; i < dao.balances.length; i++) {
-        const balance = dao.balances[i];
-        if (balance.token.symbol === "ETH") {
-          balances.push({
-            type: "native",
+      // TODO
+      // handle other tokens that are not ERC20 or eth
+      await this.ipfs.ensureOnline();
```

Comment:
If we don't do that, the calls inside the `promise.all` will call `ensureOnline` and since it is concurrent and there is only one context it never manages to get a working IPFS node, so I call the `ensureOnline` to make sure that we already have a working IPFS node so each `fetchString` does not start iterating over the nodes

Reply:
But does `toAssetBalance()` need to access IPFS?